File: problem3/problem3.c
#include <stdio.h>
#include <math.h>

int isprime(long int x) {
	if (!(x % 2)) return 0;
	for (int i = 3; i < (int)sqrt(x); i++)
		if (!(x % i)) return 0;
	return 1;
}

int main() {
	long int x = 600851475143;
	
	for(int i = sqrt(x)+1; i>0; i-=2)
		if (!(x % i) && isprime(i)) {
			printf("Result: %lli\n", i);
			return 0;
		}
}

File: problem3/problem3.py
from math import sqrt

def isprime(n):
	if n==2: return True
	if not n%2: return False
	for i in range(3, int(sqrt(n))+1, 2):
		if not n%i: return False
	return True

def primediv(n):
	for i in range(int(sqrt(n))+1, 1, -2):
		if not n%i and isprime(i):
			return i

if __name__ == '__main__':
	print('Result:', primediv(600851475143))
